src/overlays/Modal.tsx

```diff
@@ -34,20 +35,27 @@ const ModalHeader = (props: { title: string; uniqueId?: string; className?: stri
   )
 }
 
-const ModalFooter = (props: { actions: React.ReactNode[] }) => (
-  <footer className="modal__footer" data-testid="footer">
-    {props.actions.map((action: React.ReactNode, index: number) => (
-      <React.Fragment key={index}>{action}</React.Fragment>
-    ))}
-  </footer>
-)
+const ModalFooter = (props: { actions: React.ReactNode[]; className?: string }) => {
+  const classNames = ["modal__footer"]
+  if (props.className) classNames.push(props.className)
+  return (
+    <footer className={classNames.join(" ")} data-testid="footer">
+      {props.actions.map((action: React.ReactNode, index: number) => (
+        <React.Fragment key={index}>{action}</React.Fragment>
+      ))}
+    </footer>
+  )
+}
 
 export const Modal = (props: ModalProps) => {
   const uniqueIdRef = useRef(nanoid())
+  const footerClassNames = []
   const modalClassNames = ["modal"]
   const innerClassNames = ["modal__inner"]
   const closeClassNames = ["modal__close"]
   if (props.scrollableModal) innerClassNames.push("is-scrollable")
+  if (props.actionsInContent) footerClassNames.push("modal__footer-mobile")
```

Comment:
Just a nit, but this could be simplified to (props.actionsInContent) ?footerClassNames.push("modal__footer-mobile") : modalClassNames.push("modal__inner-with-footer")

Reply:
Oh wait, forgot about that closing section tag. Sorry about that! This works for me 🍭